pkg/minikube: race detected during execution of test

See https://blog.golang.org/race-detector

```
--- FAIL: TestCreateSSHShell (0.29s)
    testing.go:809: race detected during execution of test
```

```
--- FAIL: TestGetServiceURLsForService (0.00s)
    --- FAIL: TestGetServiceURLsForService/correctly_return_serviceURLs (0.00s)
        testing.go:809: race detected during execution of test
    --- FAIL: TestGetServiceURLsForService/no_host (0.00s)
        testing.go:809: race detected during execution of test
    --- FAIL: TestGetServiceURLsForService/correctly_return_empty_serviceURLs (0.00s)
        testing.go:809: race detected during execution of test
```

Current result:

```
go test -race -tags 'container_image_ostree_stub containers_image_openpgp' k8s.io/minikube/cmd/minikube/cmd k8s.io/minikube/cmd/minikube/cmd/config k8s.io/minikube/pkg/drivers k8s.io/minikube/pkg/drivers/kvm k8s.io/minikube/pkg/minikube/assets k8s.io/minikube/pkg/minikube/bootstrapper k8s.io/minikube/pkg/minikube/bootstrapper/kubeadm k8s.io/minikube/pkg/minikube/cluster k8s.io/minikube/pkg/minikube/command k8s.io/minikube/pkg/minikube/config k8s.io/minikube/pkg/minikube/cruntime k8s.io/minikube/pkg/minikube/driver k8s.io/minikube/pkg/minikube/extract k8s.io/minikube/pkg/minikube/kubeconfig k8s.io/minikube/pkg/minikube/logs k8s.io/minikube/pkg/minikube/machine k8s.io/minikube/pkg/minikube/notify k8s.io/minikube/pkg/minikube/out k8s.io/minikube/pkg/minikube/problem k8s.io/minikube/pkg/minikube/proxy k8s.io/minikube/pkg/minikube/registry k8s.io/minikube/pkg/minikube/service k8s.io/minikube/pkg/minikube/sshutil k8s.io/minikube/pkg/minikube/storageclass k8s.io/minikube/pkg/minikube/translate k8s.io/minikube/pkg/minikube/tunnel k8s.io/minikube/pkg/util k8s.io/minikube/pkg/util/lock k8s.io/minikube/pkg/util/retry
ok  	k8s.io/minikube/cmd/minikube/cmd	1.045s
ok  	k8s.io/minikube/cmd/minikube/cmd/config	0.942s
ok  	k8s.io/minikube/pkg/drivers	0.625s
ok  	k8s.io/minikube/pkg/drivers/kvm	0.615s
ok  	k8s.io/minikube/pkg/minikube/assets	0.724s
ok  	k8s.io/minikube/pkg/minikube/bootstrapper	2.328s
ok  	k8s.io/minikube/pkg/minikube/bootstrapper/kubeadm	0.861s
Stopping "mock-machine-0.68682307"...
Machine "mock-machine-0.68682307" was stopped.
Stopping "mock-machine-0.30091186"...
Machine "mock-machine-0.30091186" was stopped.
==================
WARNING: DATA RACE
Write at 0x00c0005e2770 by goroutine 102:
  bytes.(*Buffer).ReadFrom()
      /usr/lib/google-golang/src/bytes/buffer.go:200 +0x48
  io.copyBuffer()
      /usr/lib/google-golang/src/io/io.go:388 +0x3fa
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest.func1()
      /usr/lib/google-golang/src/io/io.go:364 +0xc2

Previous write at 0x00c0005e2770 by goroutine 101:
  bytes.(*Buffer).ReadFrom()
      /usr/lib/google-golang/src/bytes/buffer.go:200 +0x48
  io.copyBuffer()
      /usr/lib/google-golang/src/io/io.go:388 +0x3fa
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest.func1()
      /usr/lib/google-golang/src/io/io.go:364 +0xc2

Goroutine 102 (running) created at:
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:124 +0x7e
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleIncomingConnection()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:118 +0x1d0

Goroutine 101 (running) created at:
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:124 +0x7e
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleIncomingConnection()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:118 +0x1d0
==================
==================
WARNING: DATA RACE
Read at 0x00c0005e2750 by goroutine 102:
  bytes.(*Buffer).grow()
      /usr/lib/google-golang/src/bytes/buffer.go:73 +0x53
  bytes.(*Buffer).ReadFrom()
      /usr/lib/google-golang/src/bytes/buffer.go:202 +0x7c
  io.copyBuffer()
      /usr/lib/google-golang/src/io/io.go:388 +0x3fa
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest.func1()
      /usr/lib/google-golang/src/io/io.go:364 +0xc2

Previous write at 0x00c0005e2750 by goroutine 101:
  bytes.(*Buffer).grow()
      /usr/lib/google-golang/src/bytes/buffer.go:144 +0x23c
  bytes.(*Buffer).ReadFrom()
      /usr/lib/google-golang/src/bytes/buffer.go:202 +0x7c
  io.copyBuffer()
      /usr/lib/google-golang/src/io/io.go:388 +0x3fa
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest.func1()
      /usr/lib/google-golang/src/io/io.go:364 +0xc2

Goroutine 102 (running) created at:
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:124 +0x7e
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleIncomingConnection()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:118 +0x1d0

Goroutine 101 (running) created at:
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:124 +0x7e
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleIncomingConnection()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:118 +0x1d0
==================
==================
WARNING: DATA RACE
Read at 0x00c0005e2768 by goroutine 102:
  bytes.(*Buffer).grow()
      /usr/lib/google-golang/src/bytes/buffer.go:73 +0x79
  bytes.(*Buffer).ReadFrom()
      /usr/lib/google-golang/src/bytes/buffer.go:202 +0x7c
  io.copyBuffer()
      /usr/lib/google-golang/src/io/io.go:388 +0x3fa
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest.func1()
      /usr/lib/google-golang/src/io/io.go:364 +0xc2

Previous write at 0x00c0005e2768 by goroutine 101:
  bytes.(*Buffer).grow()
      /usr/lib/google-golang/src/bytes/buffer.go:147 +0x27f
  bytes.(*Buffer).ReadFrom()
      /usr/lib/google-golang/src/bytes/buffer.go:202 +0x7c
  io.copyBuffer()
      /usr/lib/google-golang/src/io/io.go:388 +0x3fa
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest.func1()
      /usr/lib/google-golang/src/io/io.go:364 +0xc2

Goroutine 102 (running) created at:
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:124 +0x7e
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleIncomingConnection()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:118 +0x1d0

Goroutine 101 (running) created at:
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:124 +0x7e
  k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleIncomingConnection()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/tests/ssh_mock.go:118 +0x1d0
==================
--- FAIL: TestCreateSSHShell (0.18s)
    ssh_mock.go:173: Listening on 127.0.0.1:45685
    api_mock.go:61: MockAPI.Load: &{ConfigVersion:0 Driver:0xc0000be2c0 DriverName: HostOptions:<nil> Name: RawDriver:[]} - <nil>
    driver_mock.go:46: MockDriver.GetState: Running
    ssh_mock.go:83: Accepting...
    ssh_mock.go:83: Accepting...
    ssh_mock.go:155: pty request received: &{Type:pty-req WantReply:true Payload:[0 0 0 5 120 116 101 114 109 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 6 53 0 0 0 1 0] ch:0xc0004e8000 mux:<nil>}
    ssh_mock.go:132: exec request received: &{Type:exec WantReply:true Payload:[0 0 0 4 101 120 105 116] ch:0xc0004e8000 mux:<nil>}
    testing.go:853: race detected during execution of test
FAIL
FAIL	k8s.io/minikube/pkg/minikube/cluster	0.708s
==================
WARNING: DATA RACE
Write at 0x00c000032e60 by goroutine 38:
  bytes.(*Buffer).Read()
      /usr/lib/google-golang/src/bytes/buffer.go:297 +0x4a
  bufio.(*Scanner).Scan()
      /usr/lib/google-golang/src/bufio/scan.go:213 +0x19a
  k8s.io/minikube/pkg/minikube/command.teePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner.go:198 +0x127
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix.func2()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:40 +0x83

Previous write at 0x00c000032e60 by goroutine 37:
  bytes.(*Buffer).Write()
      /usr/lib/google-golang/src/bytes/buffer.go:169 +0x42
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:46 +0x2a0
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199

Goroutine 38 (running) created at:
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:39 +0x267
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199

Goroutine 37 (running) created at:
  testing.(*T).Run()
      /usr/lib/google-golang/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/lib/google-golang/src/testing/testing.go:1204 +0xa6
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/lib/google-golang/src/testing/testing.go:1202 +0x521
  testing.(*M).Run()
      /usr/lib/google-golang/src/testing/testing.go:1119 +0x309
  main.main()
      _testmain.go:44 +0x223
==================
==================
WARNING: DATA RACE
Read at 0x00c000032e40 by goroutine 38:
  bytes.(*Buffer).Read()
      /usr/lib/google-golang/src/bytes/buffer.go:69 +0x5f
  bufio.(*Scanner).Scan()
      /usr/lib/google-golang/src/bufio/scan.go:213 +0x19a
  k8s.io/minikube/pkg/minikube/command.teePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner.go:198 +0x127
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix.func2()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:40 +0x83

Previous write at 0x00c000032e40 by goroutine 37:
  bytes.(*Buffer).grow()
      /usr/lib/google-golang/src/bytes/buffer.go:128 +0x488
  bytes.(*Buffer).Write()
      /usr/lib/google-golang/src/bytes/buffer.go:172 +0x154
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:46 +0x2a0
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199

Goroutine 38 (running) created at:
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:39 +0x267
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199

Goroutine 37 (running) created at:
  testing.(*T).Run()
      /usr/lib/google-golang/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/lib/google-golang/src/testing/testing.go:1204 +0xa6
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/lib/google-golang/src/testing/testing.go:1202 +0x521
  testing.(*M).Run()
      /usr/lib/google-golang/src/testing/testing.go:1119 +0x309
  main.main()
      _testmain.go:44 +0x223
==================
==================
WARNING: DATA RACE
Read at 0x00c000134040 by goroutine 38:
  runtime.slicecopy()
      /usr/lib/google-golang/src/runtime/slice.go:197 +0x0
  bytes.(*Buffer).Read()
      /usr/lib/google-golang/src/bytes/buffer.go:306 +0x22e
  bufio.(*Scanner).Scan()
      /usr/lib/google-golang/src/bufio/scan.go:213 +0x19a
  k8s.io/minikube/pkg/minikube/command.teePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner.go:198 +0x127
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix.func2()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:40 +0x83

Previous write at 0x00c000134047 by goroutine 37:
  runtime.slicecopy()
      /usr/lib/google-golang/src/runtime/slice.go:197 +0x0
  bytes.(*Buffer).Write()
      /usr/lib/google-golang/src/bytes/buffer.go:174 +0x115
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:48 +0x30c
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199

Goroutine 38 (running) created at:
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:39 +0x267
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199

Goroutine 37 (running) created at:
  testing.(*T).Run()
      /usr/lib/google-golang/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/lib/google-golang/src/testing/testing.go:1204 +0xa6
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/lib/google-golang/src/testing/testing.go:1202 +0x521
  testing.(*M).Run()
      /usr/lib/google-golang/src/testing/testing.go:1119 +0x309
  main.main()
      _testmain.go:44 +0x223
==================
==================
WARNING: DATA RACE
Read at 0x00c000134049 by goroutine 38:
  runtime.slicecopy()
      /usr/lib/google-golang/src/runtime/slice.go:197 +0x0
  bytes.(*Buffer).Read()
      /usr/lib/google-golang/src/bytes/buffer.go:306 +0x22e
  bufio.(*Scanner).Scan()
      /usr/lib/google-golang/src/bufio/scan.go:213 +0x19a
  k8s.io/minikube/pkg/minikube/command.teePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner.go:198 +0x127
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix.func2()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:40 +0x83

Previous write at 0x00c000134049 by goroutine 37:
  runtime.slicecopy()
      /usr/lib/google-golang/src/runtime/slice.go:197 +0x0
  bytes.(*Buffer).Write()
      /usr/lib/google-golang/src/bytes/buffer.go:174 +0x115
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:49 +0x340
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199

Goroutine 38 (running) created at:
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:39 +0x267
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199

Goroutine 37 (running) created at:
  testing.(*T).Run()
      /usr/lib/google-golang/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/lib/google-golang/src/testing/testing.go:1204 +0xa6
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/lib/google-golang/src/testing/testing.go:1202 +0x521
  testing.(*M).Run()
      /usr/lib/google-golang/src/testing/testing.go:1119 +0x309
  main.main()
      _testmain.go:44 +0x223
==================
==================
WARNING: DATA RACE
Write at 0x00c000032e58 by goroutine 38:
  bytes.(*Buffer).Read()
      /usr/lib/google-golang/src/bytes/buffer.go:307 +0x265
  bufio.(*Scanner).Scan()
      /usr/lib/google-golang/src/bufio/scan.go:213 +0x19a
  k8s.io/minikube/pkg/minikube/command.teePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner.go:198 +0x127
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix.func2()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:40 +0x83

Previous read at 0x00c000032e58 by goroutine 37:
  bytes.(*Buffer).grow()
      /usr/lib/google-golang/src/bytes/buffer.go:73 +0x79
  bytes.(*Buffer).Write()
      /usr/lib/google-golang/src/bytes/buffer.go:172 +0x154
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:46 +0x2a0
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199

Goroutine 38 (running) created at:
  k8s.io/minikube/pkg/minikube/command.TestTeePrefix()
      /usr/local/google/home/tstromberg/src/minikube/pkg/minikube/command/ssh_runner_test.go:39 +0x267
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199

Goroutine 37 (running) created at:
  testing.(*T).Run()
      /usr/lib/google-golang/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/lib/google-golang/src/testing/testing.go:1204 +0xa6
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/lib/google-golang/src/testing/testing.go:1202 +0x521
  testing.(*M).Run()
      /usr/lib/google-golang/src/testing/testing.go:1119 +0x309
  main.main()
      _testmain.go:44 +0x223
==================
--- FAIL: TestTeePrefix (0.00s)
    testing.go:853: race detected during execution of test
```

--- a/pkg/minikube/tests/ssh_mock.go
+++ b/pkg/minikube/tests/ssh_mock.go
@@ -122,11 +122,14 @@ func (s *SSHServer) handleIncomingConnection(c net.Conn) {
 
 func (s *SSHServer) handleRequest(channel ssh.Channel, req *ssh.Request) {
 	go func() {
-		if _, err := io.Copy(s.Transfers, channel); err != nil {
+		// Explicitly copy buffer contents to avoid data race
+		b := s.Transfers.Bytes()
+		if _, err := io.Copy(bytes.NewBuffer(b), channel); err != nil {
 			s.t.Errorf("copy failed: %v", err)
 		}
 		channel.Close()
 	}()
+
 	switch req.Type {
 	case "exec":
 		s.t.Logf("exec request received: %+v", req)

